cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DatastoreManager.java

```diff
@@ -333,7 +333,7 @@ private Datastore createDatastore(String dbName, KeyProvider provider) throws Da
             // if it does not exist
 
             //Pass database directory, database name, and SQLCipher key provider
-            DatastoreImpl ds = new DatastoreImpl(dbDirectory, dbName, provider);
+            QueryableDatastore ds = new QueryableDatastore(dbDirectory, dbName, provider);
```

Comment:
I don't like this because of the upfront work of the `QueryableDatastore` constructor. We shouldn't be creating the index tables etc in case they are never used.